tests/integration_tests.rs

```diff
@@ -0,0 +1,113 @@
+        .expect("kubeconfig failed to load");
+
+    let client = APIClient::new(kubeconfig);
+
+    let nodes: Api<Node> = Api::all(client.clone());
+
+    let node = nodes
+        .get("krustlet-wascc")
+        .await
+        .expect("failed to find node with name 'krustlet-wascc'");
+
+    let node_status = node.status.expect("node reported no status");
+    assert_eq!(
+        node_status
+            .node_info
+            .expect("node reported no information")
+            .architecture,
+        "wasm-wasi",
+        "expected node to support the wasm-wasi architecture"
+    );
+
+    let node_meta = node.metadata.expect("node reported no metadata");
+    assert_eq!(
+        node_meta
+            .labels
+            .expect("node had no labels")
+            .get("kubernetes.io/arch")
+            .expect("node did not have kubernetes.io/arch label"),
+        "wasm32-wascc"
+    );
+}
+
+#[tokio::test]
+async fn test_wasi_provider() {
+    // Read the environment. Note that this tries a KubeConfig file first, then
+    // falls back on an in-cluster configuration.
+    let kubeconfig = config::load_kube_config()
+        .await
+        .or_else(|_| config::incluster_config())
+        .expect("kubeconfig failed to load");
+
+    let client = APIClient::new(kubeconfig);
+
+    let nodes: Api<Node> = Api::all(client.clone());
+
+    let node = nodes
+        .get("krustlet-wasi")
+        .await
+        .expect("failed to find node with name 'krustlet-wasi'");
+
+    let node_status = node.status.expect("node reported no status");
+    assert_eq!(
+        node_status
+            .node_info
+            .expect("node reported no information")
+            .architecture,
+        "wasm-wasi",
+        "expected node to support the wasm-wasi architecture"
+    );
+
+    let node_meta = node.metadata.expect("node reported no metadata");
+    assert_eq!(
+        node_meta
+            .labels
+            .expect("node had no labels")
+            .get("kubernetes.io/arch")
+            .expect("node did not have kubernetes.io/arch label"),
+        "wasm32-wasi"
+    );
+
+    let pods: Api<Pod> = Api::namespaced(client.clone(), "default");
+    let p: serde_json::Value = json!({
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": {
+            "name": "helloworld"
+        },
+        "spec": {
+            "containers": [
+                {
+                    "name": "helloworld",
+                    "image": "bacongobbler.azurecr.io/helloworld:v1.0.0",
+                },
+            ],
+            "nodeSelector": {
+                "beta.kubernetes.io/arch": "wasm32-wasi",
+            },
+        }
+    });
+    let pod_data = serde_json::to_vec(&p).expect("pod should always be serializable to JSON");
+
+    let pod = pods
+        .create(&PostParams::default(), pod_data)
+        .await
+        .expect("could not create pod");
+
+    assert_eq!(pod.status.unwrap().phase.unwrap(), "Pending");
```

Comment:
We definitely should get past Pending here. It should probably be Running (and updated to `Completed` once we get status working)

Reply:
This is just an initial submission. Happy to go further here but I figured I should stop somewhere before I go down the whole rabbit hole. A few tests were failing previously so I wanted to incrementally add more tests in follow-up PRs.

Reply:
Added a comment as a TODO item for follow-up